### ISSUE TITLE
[BUG][jaxrs-spec] enumUnknownDefaultCase not deserialized

### ISSUE THREAD

**Issue by PhilippParis** [OpenAPITools/openapi-generator#13444]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

using the jaxrs-spec (and probably other jaxrs) generator the  'enumUnknownDefaultCase' property is not working correctly:
altough the additional enum value is created, the value is not used in the deserialization of the enum

###### Expected Output
````java
  @JsonCreator
  public static ColorDto fromValue(String value) {
    for (ColorDto b : ColorDto.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    return UNKNOWN_DEFAULT_OPEN_API;
  }
````
###### Actual Output
````java
  @JsonCreator
  public static ColorDto fromValue(String value) {
    for (ColorDto b : ColorDto.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
````

##### openapi-generator version

6.1.0

##### OpenAPI declaration file content or url

```yaml
openapi: "3.0.0"
info:
  version: 2.0.0
  title: test
paths:
  /pets:
    get:
      summary: List all pets
      operationId: listPets
      responses:
        '200':
          description: OK
components:
  schemas:
    Color:
      type: string
      enum: [RED, BLUE, GREEN]
```

##### Generation Details

using the openapi-maven-generator plugin with configOption
````
<enumUnknownDefaultCase>true</enumUnknownDefaultCase>
````

##### Steps to reproduce

generate code

##### Related issues/PRs

In https://github.com/OpenAPITools/openapi-generator/issues/11886 the same problem was fixed for the 'java' generator

##### Suggest a fix

in 'modelEnum.class', 'enumClass.mustache' and 'enumOuterClass.mustache' in resources/JavaJaxRs **and all subfolders**:

replace
````
{{#isNullable}}return null;{{/isNullable}}{{^isNullable}}throw new IllegalArgumentException("Unexpected value '" + value + "'");{{/isNullable}}
````

with 
````
{{#isNullable}}return null;{{/isNullable}}{{^isNullable}}{{#enumUnknownDefaultCase}}{{#allowableValues}}{{#enumVars}}{{#-last}}return {{{name}}};{{/-last}}{{/enumVars}}{{/allowableValues}}{{/enumUnknownDefaultCase}}{{^enumUnknownDefaultCase}}throw new IllegalArgumentException("Unexpected value '" + value + "'");{{/enumUnknownDefaultCase}}{{/isNullable}}
````

**Comment by AhmedAbdelfaheem** [comment]
@DeaneOC  Any updates ?

**Comment by Newmmen** [comment]
@DeaneOC Any updates ?

**Comment by nahogft** [comment]
Is there any update? which version of openapi generator has already the fix?

**Comment by nahogft** [comment]
Is there any update? which version of openapi generator has already the fix?

**Comment by gijsdh** [comment]
Any updates on this?

**Comment by wing328** [comment]
merged https://github.com/OpenAPITools/openapi-generator/pull/18910

please pull the latest master to give it a try

**Comment by wing328** [comment]
can you please file a PR and we will review accordingly?

**Comment by nahogft** [comment]
Is basically the same changes but applied on JavaSpring folder

**Comment by patrick-solving-problems** [comment]
> merged [#18910](https://github.com/OpenAPITools/openapi-generator/pull/18910)
> 
> please pull the latest master to give it a try

I am using 7.12.0 and it works fine for me. Thanks for your work!
